Classes/DemandProvider/TypoScriptDemandProvider.php

```diff
@@ -5,8 +5,16 @@
 
 namespace Pixelant\Demander\DemandProvider;
 
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use \TYPO3\CMS\Extbase\Configuration\ConfigurationManager;
+
 
 class TypoScriptDemandProvider implements DemandProviderInterface
 {
-
-}
\ No newline at end of file
+    public function getDemand(): array
+    {
+        $configurationManager = GeneralUtility::makeInstance(ConfigurationManager::class);
+        $config = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT)['config.']['tx_demander.'];
+        return $config['demands.'];
```

Comment:
1. You have to process the array so it returns compatible arrays. The keys for array values will have dots at the end, and they need to be removed. Just like `$config['demands.']` has a dot, there's going to be a `$config['demands.']['tablename-fieldname.']`, but we can't have the dot in the `tablename-fieldname.` key.

2. Return an empty array if there is no configuration to make sure this method never returns null.